### PR TITLE
Added support for models trained in python and brainscript

### PR DESCRIPTION
**Pull request by aribornstein**

### CODE REVIEW

File: frcnn_detector.py [hunk 1]
```diff
@@ -18,13 +18,13 @@
 roi_maxDimRel = 0.4
 roi_minNrPixelsRel = 2 * roi_minDimRel * roi_minDimRel
 roi_maxNrPixelsRel = 0.33 * roi_maxDimRel * roi_maxDimRel
-roi_maxAspectRatio = 4.0  # maximum aspect Ratio of a ROI vertically and horizontally
+roi_maxAspectRatio = 7.0  # maximum aspect Ratio of a ROI vertically and horizontally
```
Comment (nadavbar) [inline frcnn_detector.py]: Is this related to the change? If it fits a certain scenario I rather think of a better way that we can set this as parameter.

File: frcnn_detector.py [hunk 2]
```diff
@@ -18,13 +18,13 @@
...
 roi_maxNrPixelsRel = 0.33 * roi_maxDimRel * roi_maxDimRel
-roi_maxAspectRatio = 4.0  # maximum aspect Ratio of a ROI vertically and horizontally
+roi_maxAspectRatio = 7.0  # maximum aspect Ratio of a ROI vertically and horizontally
 roi_maxImgDim = 200  # image size used for ROI generation
 ss_scale = 100  # selective search ROIS: parameter controlling cluster size for segmentation
 ss_sigma = 1.2  # selective search ROIs: width of gaussian kernal for segmentation
 ss_minSize = 20  # selective search ROIs: minimum component size for segmentation
 grid_nrScales = 7  # uniform grid ROIs: number of iterations from largest possible ROI to smaller ROIs
-grid_aspectRatios = [1.0, 2.0, 0.5]  # uniform grid ROIs: aspect ratio of ROIs
+grid_aspectRatios = [0.5, 4.0, 6.0]  # uniform grid ROIs: aspect ratio of ROIs
```
Comment (nadavbar) [inline frcnn_detector.py]: Same comment above

File: frcnn_detector.py [hunk 3]
```diff
@@ -61,6 +61,7 @@ def __init__(self, model_path,
         self.__use_selective_search_rois = use_selective_search_rois
         self.__use_grid_rois = use_grid_rois
         self.__model = None
+        self.__isPythonModel = None
```
Comment (nadavbar) [inline frcnn_detector.py]: Why None?
If it's a boolean I'd rather initialize it to True/False. 
It's also more efficient

File: frcnn_detector.py [hunk 4]
```diff
@@ -93,10 +94,17 @@ def warm_up(self):
         dummy_image = np.ones((3, self.__resize_width, self.__resize_height)) * 255.0
 
         # prepare the arguments
-        arguments = {
-            self.__model.arguments[self.__args_indices["features"]]: [dummy_image],
-            self.__model.arguments[self.__args_indices["rois"]]: [dummy_rois]
-        }
+        if (self.__isPythonModel):#python model
```
Comment (nadavbar) [inline frcnn_detector.py]: Remove the comment at the end of the line. It's pretty clear what _isPythonModel means :)

File: frcnn_detector.py (hunk 3 above)
Comment (nadavbar) [inline frcnn_detector.py]: In general for python I use (or try my best to use) PEP8 notation, which means lower case variable names and underscores as seperators. So __isPythonModel should be changed to __is_python_model

File: frcnn_detector.py [hunk 5]
```diff
@@ -107,39 +115,48 @@ def load_model(self):
...
-        roi_input = input_variable(rois_node.shape, name='rois')
-
-        # Clone the desired layers with fixed weights and place holder for the new input nodes
-        cloned_nodes = combine([z_node.owner]).clone(
-            CloneMethod.freeze,
-            {features_node: placeholder(name='features'), rois_node: placeholder(name='rois')})
-
-        # apply the cloned nodes to the input nodes to obtain the model for evaluation
-        self.__model = cloned_nodes(image_input, roi_input)
+        self.__isPythonModel = (True if (len(trained_frcnn_model.arguments) < 3) else False)
```
Comment (nadavbar) [inline frcnn_detector.py]: I think that the brackets at the start and end might not be needed

File: frcnn_detector.py [hunk 6]
```diff
@@ -107,39 +115,48 @@ def load_model(self):
...
+
+        if (self.__isPythonModel):#python model
+            self.__nr_rois = trained_frcnn_model.arguments[1].shape[0]
+            self.__resize_width = trained_frcnn_model.arguments[0].shape[1]
+            self.__resize_height = trained_frcnn_model.arguments[0].shape[2]
+            self.labels_count = 4 # this should be the number of labels in the model
```
Comment (nadavbar) [inline frcnn_detector.py]: Why 4? You should take it from the model

File: frcnn_detector.py [hunk 7]
```diff
@@ -262,17 +279,27 @@ def detect(self, img):
         dummy_labels = np.zeros((self.__nr_rois, self.labels_count))
 
         # prepare the arguments
-        arguments = {
-            self.__model.arguments[self.__args_indices["features"]]: [img_model_arg],
-            self.__model.arguments[self.__args_indices["rois"]]: [test_rois]
-        }
+        if (self.__isPythonModel):#python model
```
Comment (nadavbar) [inline frcnn_detector.py]: Instead of doing this if, you can just set the values of self.__args_indices["features"] and self.__args_indices["rois"] when you load the python model and then this code path can remain unchanged.

File: frcnn_detector.py [hunk 8]
```diff
@@ -262,17 +279,27 @@ def detect(self, img):
...
 
         # run it through the model
         output = self.__model.eval(arguments)
         self.__model_warm  = True
         
         # take just the relevant part and cast to float64 to prevent overflow when doing softmax
-        rois_values = output[0][0][:roi_padding_index].astype(np.float64)
+        if (self.__isPythonModel):#python model
+            rois_values = output[0][:roi_padding_index].astype(np.float64)
+            print("===", rois_values.shape)
```
Comment (nadavbar) [inline frcnn_detector.py]: You can remove this print

**Review by nadavbar — CHANGES_REQUESTED** [review]
Hi, overall looks OK, but needs a few fixes which are mentioned in the comments.

File: frcnn_detector.py (hunk 1 above)
Comment (aribornstein) [reply frcnn_detector.py]: Agreed for now i'll change it back
